internal/encrypt/crypt.go

```diff
@@ -0,0 +1,233 @@
+package encrypt
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/inconshreveable/log15"
+)
+
+const (
+	validKeyLength    = 32
+	hmacSize          = sha256.Size
+	primaryKeyIndex   = 0
+	secondaryKeyIndex = 1
+)
+
+type EncryptionError struct {
+	Message string
+	Err     error
+}
+
+func (e *EncryptionError) Error() string { return e.Message + ": " + e.Err.Error() }
+func (e *EncryptionError) Unwrap() error { return e.Err }
+
+// ErrNotEncoded means we can test for whether or not a string is encoded, prior to attempting decryption
+var ErrNotEncoded = errors.New("object is not encoded")
+
+// Generate a valid key for AES-256 encryption
+func GenerateRandomAESKey() ([]byte, error) {
+	b := make([]byte, validKeyLength)
+	_, err := rand.Read(b)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+// EncryptBytesIfPossible encrypts the byte array if encryption is configured.
+// Returns an error only when encryption is enabled, and encryption fails.
+func EncryptBytesIfPossible(b []byte) (string, error) {
```

Comment:
```suggestion
// EncryptBytes encrypts the bytes and returns the string representation.
// An error is returned when encryption fails. It returns the original 
// bytes if encryption is not configured.
func EncryptBytes(b []byte) (string, error) {
```

I suggest to remove `IfPossible` part because there is no method would fail when encryption is not configured, which seems to me the caller shouldn't care whether encryption is configured, this is the package's internal concern. WDYT?

Reply:
Was this package renamed to `encrypt`? I think `secrets` might be better since I think `secrets.Encrypt` and `secrets.Decrypt` read better than `encrypt.Encrypt`, `encrypt.Decrypt`.

Reply:
I also think we shouldn't use the `IfPossible` pattern but rather have something like this:
- One `Encryptor` interface
- One `encryptor` implementation: the actual encryption logic
- One `noOpEncryptor` implementation: does nothing
Depending on when the encryption is configured you can set one or the other

Reply:
Refactored using interface approach suggested by @asdine.